tools/test_containers_compose

```diff
@@ -2,11 +2,53 @@
 
 set -euo pipefail
 
-for i in webui worker; do
+wait_until(){
+  command=$1
+  timeout=$2
+
+  i=1
+  until eval "$command"
+  do
+      ((i++))
+      if [[ $i -gt $timeout ]]; then
+          echo "Command '$command' aborted because of timeout"
+          exit 1
+      fi
+      sleep 1
+  done
+}
+
+setup_containers(){
   (
-    cd container/$i/ &&
-     for retry in {2..0}; do sudo docker-compose build && break; echo "Remaining retries $retry"; done &&
-    sudo docker-compose up -d &&
+    for retry in {2..0}; do sudo docker-compose build && break; echo "Remaining retries $retry"; done &&
+    sudo MOJO_CLIENT_DEBUG=1 docker-compose up -d &&
     (docker-compose ps --services --filter status=stopped | grep "^[[:space:]]*$") || (docker-compose logs; sudo docker-compose ps; exit 1)
   ) || exit 1;
-done
+}
+
+test_webui(){
+  (
+    workspace=$(mktemp -d) &&
```

Comment:
please check that the temporary directory will be removed again on both passed as well as failed test runs. This is why I suggested the "trap".

Reply:
done